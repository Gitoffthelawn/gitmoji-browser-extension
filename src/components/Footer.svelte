<script>
  import { page } from '../stores'
</script>

<footer>
  <nav>
    <button
      on:click={() => {
        page.goTo('settings')
      }}
    >
      settings
    </button>
    <a
      href="https://github.com/johannchopin/gitmoji-browser-extension"
      target="_blank"
      class="github"
      rel="noreferrer"
    >
      GitHub
    </a>
  </nav>
</footer>

<style>
  footer {
    position: relative;
    z-index: 1;
  }

  :global(body.dark) footer {
    background-color: black;
  }

  a {
    color: black;
    text-decoration: none;
  }

  :global(body.dark) a,
  :global(body.dark) button {
    color: white;
  }

  nav {
    display: flex;
    justify-content: center;
    padding: 0.25em 0;
  }

  nav > *:not(:last-child):after {
    content: "â€¢";
    margin: 0 0.2em;
  }

  button {
    border: none;
    padding: 0;
    background-color: transparent;
    font-size: 1em;
    cursor: pointer;
  }

  .github {
    margin-bottom: -0.2em; /* Little trick to center the link verticaly */
  }
</style>
