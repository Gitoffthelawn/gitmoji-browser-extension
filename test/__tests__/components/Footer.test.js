import { render } from '@testing-library/svelte'
import Footer from '../../../src/components/Footer.svelte'

describe('<Footer />', () => {
  let wrapper

  beforeEach(() => {
    wrapper = render(Footer)
  })

  it('should match snapshot', () => {
    const { container } = wrapper

    expect(container).toMatchSnapshot()
  })
})
