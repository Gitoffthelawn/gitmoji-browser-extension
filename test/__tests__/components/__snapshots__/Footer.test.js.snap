// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<body>
  <div>
    <footer>
      <nav>
        <button>
          settings
        </button>
         
        <a
          class="github"
          href="https://github.com/johannchopin/gitmoji-browser-extension"
          target="_blank"
        >
          GitHub
        </a>
      </nav>
    </footer>
  </div>
</body>
`;
